File: 백준/Silver/3273. 두 수의 합/두 수의 합.cc
#include <bits/stdc++.h>
using namespace std;

long long n, x, a[100004], ret;
int main() {
  cin >> n;
  for (int i = 0; i < n; i++) cin >> a[i];
  cin >> x;
  int l = 0, r = n - 1;

  sort(a, a + n);

  while (l < r) {
    if (a[l] + a[r] == x) {
      ret++;
      l++;
      r--;
    } else if (a[l] + a[r] < x) {
      l++;
    } else {
      r--;
    }
  }
  cout << ret;
  return 0;
}

File: 백준/Silver/3273. 두 수의 합/두 수의 합.js
const fs = require('fs');
const input = fs.readFileSync('/dev/stdin').toString().trim().split('\n');

const n = Number(input[0]);
const arr = input[1].split(' ').map(e => Number(e));
const x = Number(input[2]);

function answer (n, arr, target){
    let ret = 0;

    arr.sort((a, b) => a - b);
    // console.log(arr);

    let l = 0, r = arr.length - 1;
    
    while(l < r){
        if(arr[l] + arr[r] > target){
            r--;
        }else if(arr[l] + arr[r] < target){
            l++;
        }else {
            ret++; 
            l++;
        }
    }
    
    return ret;
}

console.log(answer(n, arr, x));